packages/lesswrong/components/editor/EditorFormComponent.tsx

```diff
@@ -72,6 +72,19 @@ export const EditorFormComponent = ({form, formType, formProps, document, name,
       }
     }
   }, [getLocalStorageHandlers, currentEditorType]);
+
+  /**
+   * Update the edited field (e.g. "contents") so that other form components can access the updated value. The direct motivation for this
+   * was for SocialPreviewUpload, which needs to know the body of the post in order to generate a preview description and image.
+   */
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const throttledSetContentsValue = useCallback(_.throttle(async () => {
+    if (!(editorRef.current && shouldSubmitContents(editorRef.current))) return
+    
+    // Preserve other fields in "contents" which may have been sent from the server
+    updateCurrentValues({[fieldName]: {...(document[fieldName] || {}), ...(await editorRef.current.submitData(null))}})
+  }, autosaveInterval, {leading: true}),
+  [saveBackup, autosaveInterval])
```

Comment:
I think this is reasonable - I didn't notice any slowdown when typing, plus when using collab editing, things don't get saved in local storage anyway. Though I was testing with a collab editing doc and the preview image doesn't get populated properly when I open the editor to edit the doc (though I think it does get populated when the text changes).

Also I noticed this issue with the top bar showing too many things, which seems to not replicate on `master` but I'm not totally convinced is related to this branch. I saw it when I first shared the doc, and then later I saw it when I opened the share link in incognito and typed some things as a second user, then closed that window.
<img width="678" alt="Screen Shot 2023-01-27 at 11 16 01 AM" src="https://user-images.githubusercontent.com/9057804/215137394-2b393ca6-7e39-466a-ab4a-842470bdd400.png">

Reply:
Thanks for catching that, it turned out to be a fairly nasty bug that I've had to do a workaround for, see:
```
  /**
   * This is a workaround for a bug resulting from the PresenceList and TrackChangesData plugins.
   * When getDataWithDiscardedSuggestions is called in Editor.tsx (called via throttledSetContentsValue)
   * it results in a new presence list being added with "0 connected users". I have no idea why this happens,
   * and both the plugins are closed source so it will be very hard to work out. This workaround removes all
   * but the first presence list, which is the one that actually contains the correct data. Any others that
   * are created temporarily are also hidden by the "& .ck-presence-list:nth-child(n+2)" selector above.
   */
  const presenceListChildren = presenceListRef.current?.querySelectorAll(".ck-presence-list");
  useEffect(() => {
    // remove all but the first presence list
    if (!presenceListChildren) return;

    const [_, ...otherChildren] = Array.from(presenceListChildren);
    for (let child of otherChildren) {
      child.remove();
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [presenceListChildren?.length])
```

Reply:
Oof